framework/src/outputs/Checkpoint.C

```diff
@@ -248,3 +230,69 @@ Checkpoint::updateCheckpointFiles(CheckpointFileNames file_struct)
                             _problem_ptr->mesh().isDistributedMesh() ? comm().size() : 1);
   }
 }
+
+void
+Checkpoint::validateExecuteOn() const
+{
+  const auto & execute_on = getParam<ExecFlagEnum>("execute_on");
+
+  // Create a vector containing all valid values of execute_on
+  std::vector<ExecFlagEnum> valid_execute_on_values(7);
+  {
+    ExecFlagEnum valid_execute_on_value = execute_on;
+    valid_execute_on_value = {EXEC_INITIAL};
+    valid_execute_on_values[0] = valid_execute_on_value;
+    valid_execute_on_value = {EXEC_TIMESTEP_END};
+    valid_execute_on_values[1] = valid_execute_on_value;
+    valid_execute_on_value = {EXEC_FINAL};
+    valid_execute_on_values[2] = valid_execute_on_value;
+    valid_execute_on_value = {EXEC_INITIAL, EXEC_TIMESTEP_END};
+    valid_execute_on_values[3] = valid_execute_on_value;
+    valid_execute_on_value = {EXEC_TIMESTEP_END, EXEC_FINAL};
+    valid_execute_on_values[4] = valid_execute_on_value;
+    valid_execute_on_value = {EXEC_INITIAL, EXEC_FINAL};
+    valid_execute_on_values[5] = valid_execute_on_value;
+    valid_execute_on_value = {EXEC_INITIAL, EXEC_TIMESTEP_END, EXEC_FINAL};
+    valid_execute_on_values[6] = valid_execute_on_value;
+  }
+
```

Comment:
```
const auto & execute_on = getParam<ExecFlagEnum>("execute_on");
const std::set<ExecFlagType> allowed = {EXEC_INITIAL, EXEC_TIMESTEP_END, EXEC_FINAL};
for (const auto & value : execute_on)
  if (!allowed.count(value))
    paramError("execute_on", "The exec flag ", value, " is not allowed... etc");
```

I know that this is old code too, but this is probably the better option

Reply:
Beautiful.